overridable/overridable.go

```diff
@@ -64,6 +77,19 @@ func (r rules) Match(name string) interface{} {
 	return nil
 }
 
+// MatchWithSuffix matches the given repository name against all rules and the
+// suffix against provided pattern suffix, returning the rule value that matches
+// at last, or nil if none match.
+func (r rules) MatchWithSuffix(name, suffix string) interface{} {
+	// We want the last match to win, so we'll iterate in reverse order.
+	for i := len(r) - 1; i >= 0; i-- {
+		if r[i].compiled.Match(name) && (r[i].patternSuffix == "" || r[i].patternSuffix == suffix) {
```

Comment:
would we want to support globbing in the suffix too?

Reply:
I'm not convinced it's worth it at the moment. For our usecase I don't see, for example, someone publishing a N>1 subset of changesets in a repository where manually listing the branch names wouldn't be feasible. But if we play around with it and notice that it's something we want, we can add it.

Reply:
My suspicion is that we'll eventually want it, but I think it's a lot better for us to start without suffix globbing and see how far our users get first.